templates/minimal/node/src/command.rs

```diff
@@ -51,12 +56,34 @@ impl SubstrateCli for Cli {
 		2017
 	}
 
-	fn load_spec(&self, id: &str) -> Result<Box<dyn sc_service::ChainSpec>, String> {
-		Ok(match id {
-			"dev" => Box::new(chain_spec::development_config()?),
-			path =>
-				Box::new(chain_spec::ChainSpec::from_json_file(std::path::PathBuf::from(path))?),
-		})
+	fn load_spec(&self, maybe_path: &str) -> Result<Box<dyn sc_service::ChainSpec>, String> {
+		// TODO: it would be good to be able to provide a chain-spec that has no code, and code is
+		// still read from `--runtime`. So your chain spec can be cleaner, and it is only initial
+		// state.
+		let code = std::fs::read(&self.runtime)
+			.map_err(|e| format!("Failed to runtime read {}: {}", &self.runtime, e))?;
+
+		Ok(Box::new(if maybe_path.is_empty() {
+			println!("Using development chain spec, no genesis state set.");
+			let mut properties = Properties::new();
+			properties.insert("tokenDecimals".to_string(), 0.into());
+			properties.insert("tokenSymbol".to_string(), "MINI".into());
+
+			// `.with_genesis_config(Default::default)` won't work, but should.
```

Comment:
@michalkucharczyk this was my main chain-spec related issue so far, I would have expected it to work without this. Or perhaps the `Default::default` should do this for me.

Reply:
Actually, `.with_genesis_config_patch(Default::default)` should do the job. (which means empty patch for default config).